pkgs/development/libraries/libgphoto2/default.nix

```diff
@@ -16,17 +18,17 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-c7fBl6GBLAU+RL5WFC4PL+n/nEHZUfqIJ9qq1+qNNCg=";
   };
 
-  patches = [];
+  depsBuildBuild = [ pkg-config ];
 
   nativeBuildInputs = [
     autoreconfHook
-    pkg-config
     gettext
     libtool
   ];
 
   buildInputs = [
     libjpeg
+    libtool
```

Comment:
Might be useful to have this fact also in comment.

Reply:
You don't think it would be redundant?  It'll be pretty obvious that it's required as soon as anybody tries to remove it (assuming they try cross compiling, which I'd hope they would when removing a dependency shared between buildInputs and nativeBuildInputs…).

Reply:
Personally, I tend to add a comment mentioning the specific component that is required when a package contains multiple components. Doubly so when the package is primarily used as a build-time tool so using it at runtime is contrary to expectations. For example, I completely forgot libltdl is a thing.

But when someone gets confused, they can always check blame (great commit message) so I would not say the comment is absolutely necessary.